app/models/secret.rb

```diff
@@ -54,6 +54,10 @@ def update_expiration(resource_id, expires_at)
       Secret.where(resource_id: resource_id).update(expires_at: expires_at)
     end
 
+    def current_values_for_update(resource_ids)
+      Secret.for_update.select(resource_id: resource_ids).all
```

Comment:
It looks like this is only used in the commented out method `conjur_facade.rb#lock_secrets_for_resource_ids`, but...

Can you elaborate on the purpose of of this method? I ask because I don't believe this use of `for_update` is doing what you expect. When you end the chain with `.all`, this is going to execute the command with the update lock and return the results as an Array, with the lock released.

If you want downstream operations to use the lock, you should be returning the dataset (no `.all`) rather than the result array.